mido/midifiles/midifiles.py

```diff
@@ -70,6 +88,37 @@ def _dbg(text=''):
     print(text)
 
 
+def _handle_error(error_msg, error_mode):
```

Comment:
For example:

```python
import warnings

# Let's do our own warning type so it can be ignored without ignoring other modules warnings.
class MidiFileWarning(Warning):
    pass

# Use this whenever you want to display errors, instead of `_handle_error`.
warnings.warn('The error message', MidiFileWarning)
```

Then, either have a public API for setting the warning level:
```python
def set_error_level(error_mode):
    if error_mode == 'raise':
        warnings.filterwarnings('error', category=MidiFileWarning)
    elif error_mode == 'warn':
        warnings.filterwarnings('default', category=MidiFileWarning)
    elif error_mode == 'ignore':
        warnings.filterwarnings('ignore', category=MidiFileWarning) 
    else:
        raise ValueError('unknown error mode {!r}'.format(error_mode))
```
Or, preferably skip the if statements and just document that the user can call `warnings.filterwarnings(action, category=MidiFileWarning)` themselves!

Note that warnings introduces a global state per Python process so to be really correct, each function should use the context manager `warnings.catch_warnings` and take the `error_mode` argument, for example:

```python
def read_message(infile, status_byte, peek_data, delta, error_mode):
    # Stuff...

    # We want to print a warning with error_mode, so we temporarily change the warnings filters according to error_mode with a context manager.
    with warnings.catch_warnings():
        warnings.filterwarnings(error_mode, category=MidiFileWarning)
        warnings.warn('The error message', MidiFileWarning)

    # More stuff...
```

That might be a little overkill initially though, so perhaps just skip having `error_mode` entirely and leave it as a global state. Unless multiprocessing is expected, of course. :confounded: